Framework/Algorithms/test/ScaleXTest.h

```diff
@@ -110,20 +114,17 @@ class ScaleXTest : public CxxTest::TestSuite {
 
       for (size_t j = 0; j < xsize; ++j) {
         if (factor > 0) {
-          TS_ASSERT_DELTA(result->readX(i)[j], factor * inputWS->readX(i)[j],
-                          1e-12);
-          TS_ASSERT_EQUALS(result->readY(i)[j], inputWS->readY(i)[j]);
-          TS_ASSERT_EQUALS(result->readE(i)[j], inputWS->readE(i)[j]);
+          TS_ASSERT_DELTA(result->x(i)[j], factor * inputWS->x(i)[j], 1e-12);
```

Comment:
To make these more efficient you can take do something like `auto &X = result->x(i);` then use X (or Y/E) instead of accessing the reference in a tight loop over all spectra. Check other places where this occurs.